src/Step.js

```diff
@@ -3,32 +3,49 @@ import { useEffect, createContext, useContext } from 'react';
 
 export const StepContext = createContext<Object>(null);
 
-type Props = {
+type Props = {|
   children: React$Node,
-} & Losen$Step;
-
-const Step = ({ children, name, validator, autoSkip }: Props) => {
-  const { registerStep, activeStep, updateStep, initialized } = useContext(
-    StepContext,
-  );
-
-  const stepInfo = {
-    name,
-    validator,
-    autoSkip,
-  };
+  ...Losen$Step,
+|};
+
+const Step = ({ children, name, validator, autoSkip, state }: Props) => {
+  const {
+    registerStep,
+    activeStep,
+    updateStep,
+    initialized,
+    stateManager,
+  } = useContext(StepContext);
 
   useEffect(() => {
     if (!initialized) {
-      registerStep(stepInfo);
+      registerStep({
+        name,
+        validator,
+        autoSkip,
+      });
     }
-  }, [name]);
+  }, [name, validator, autoSkip, initialized, registerStep]);
 
   useEffect(() => {
     if (initialized) {
-      updateStep(stepInfo);
+      updateStep({
+        name,
+        validator,
+        autoSkip,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [autoSkip, validator, initialized]);
```

Comment:
During my refactor (https://github.com/otovo/react-losen/pull/25/commits/d1fcbeb896a19e16ac372bb142986798919f08af) I noticed that `name` is missing from this dependency list. It's super easy to miss bugs like this when we ignore linting rules 😅